### ISSUE TITLE
GMR: cannot return grouped stream without a reduction

### ISSUE THREAD

**Issue by coffeemug** [rethinkdb/rethinkdb#2025]
```
r.table('wiki_articles')('text').group(r.row.split())
```

Output:

```
RqlRuntimeError: Cannot return grouped stream without a reduction (`reduce`, `sum`, etc.) on the end.
```

I discovered this to be spectacularly annoying in practice for all of the following reasons:
- During development it's really nice to see intermediary results, so I basically always tack on `coerceTo('array')` at the end
- I can't even tack on `toArray()` since we don't have that! It's `coerceTo('array')`!! (err!)
- The error message doesn't mention that we can run `coerceTo('array')`, so users won't be able to discover it. Also `groupsToArray` doesn't work on grouped streams.

TL;DR: I think we should bite the bullet and call `coerceTo('array')` when returning grouped streams implicitly. (There is, admittedly potential for blowing RAM because our array size limit doesn't deal too well with this case, but I feel like that's ok -- we should just handle it via #1375).
